### ISSUE TITLE
NullReferenceException on NaiveBayes Learn

### ISSUE THREAD

**Issue by souserge** [accord-net/framework#758]
Hi, when trying to train Naive Bayes on multi-label data, I receive a NullReferenceException. Is it just because the method supports only one-dimentional vector of data?
Here's the code I'm using (the exception is thrown on the last statement):

```
var X = new double[][]
    {new double[] {1, 0, 1, 1, 0},
     new double[] {0, 1, 0, 0, 1},
     new double[] {1, 0, 1, 0, 1}};

var Y = new double[][]
    {new double[] {1, 0},
     new double[] {0, 1},
     new double[] {1, 1}};

var teacher = new NaiveBayesLearning<NormalDistribution>();
var bayes = teacher.Learn(X, Y);
```

**Comment by cesarsouza** [comment]
Hi @souserge,

Many thanks for opening the issue. The problem seems to derive from the current implementation for multi-label problems, which was not kept up-to-date with the implementation for multi-class classification problems. I might be able to commit a fix in the following hours.

Regards,
Cesar

**Comment by cesarsouza** [comment]
A new pre-release package (3.6.4-alpha) has just been uploaded to NuGet. It should contain a fix for this issue.

**Comment by souserge** [comment]
Thank You very much, It works. 
Maybe it's a bit off-top, but it seems that the discrete Naive Bayes doesn't support multi-label classification (i.e. like with data in the example above without any specified distribution). Is it a bug of a feature?

**Comment by cesarsouza** [comment]
@souserge, the Naive Bayes available in the latest pre-release (3.6.4-alpha) should have include support for multi-label classification for both discrete and real/generic implementations (I actually only checked the current development version and not 3.6.4-alpha, but I am under the impression I didn't do changes to Naive Bayes since then).

May I ask which error are you getting? Do you have an example of the code you are using?

**Comment by souserge** [comment]
@cesarsouza, it excuse me for bothering you, for some reason, I couldn't pass an int[][] array of class labels to Teacher.Learn(). Reinstalled the package and the issue has vanished.
